chips/rp2040/src/gpio.rs

```diff
@@ -347,13 +347,13 @@ impl<'a> RPPins<'a> {
                 let l_low_reg_no = pin * 4;
                 if (current_val & enabled_val & (1 << l_low_reg_no)) != 0 {
                     self.pins[pin + bank_no * 8].handle_interrupt();
-                } else if (current_val & enabled_val & (1 << l_low_reg_no + 1)) != 0 {
+                } else if (current_val & enabled_val & (1 << (l_low_reg_no + 1))) != 0 {
                     self.pins[pin + bank_no * 8].handle_interrupt();
-                } else if (current_val & enabled_val & (1 << l_low_reg_no + 2)) != 0 {
-                    self.gpio_registers.intr[bank_no].set(current_val & (1 << l_low_reg_no + 2));
+                } else if (current_val & enabled_val & (1 << (l_low_reg_no + 2))) != 0 {
+                    self.gpio_registers.intr[bank_no].set(current_val & (1 << (l_low_reg_no + 2)));
                     self.pins[pin + bank_no * 8].handle_interrupt();
-                } else if (current_val & enabled_val & (1 << l_low_reg_no + 3)) != 0 {
-                    self.gpio_registers.intr[bank_no].set(current_val & (1 << l_low_reg_no + 3));
+                } else if (current_val & enabled_val & (1 << (l_low_reg_no + 3))) != 0 {
+                    self.gpio_registers.intr[bank_no].set(current_val & (1 << (l_low_reg_no + 3)));
                     self.pins[pin + bank_no * 8].handle_interrupt();
                 }
```

Comment:
The lack of parallelism here for the `(l_low_reg_no + 1)` case feels like a bug.

Reply:
Moved to issue #3659